src/state.rs

```diff
@@ -0,0 +1,372 @@
+                BevyWindowEvent::FileDragAndDrop(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::Ime(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::RequestRedraw(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowBackendScaleFactorChanged(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowCloseRequested(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowCreated(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowDestroyed(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowFocused(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowMoved(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowOccluded(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowResized(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowScaleFactorChanged(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::WindowThemeChanged(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::MouseButtonInput(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::MouseMotion(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::MouseWheel(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::PinchGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::RotationGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::DoubleTapGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::PanGesture(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::TouchInput(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::KeyboardInput(e) => {
+                    world.send_event(e);
+                }
+                BevyWindowEvent::KeyboardFocusLost(e) => {
+                    world.send_event(e);
+                }
+            }
+        }
+
+        if !buffered_events.is_empty() {
+            world
+                .resource_mut::<Events<BevyWindowEvent>>()
+                .send_batch(buffered_events);
+        }
+    }
+}
+
+impl CompositorHandler for SmithayRunnerState {
```

Comment:
Do we need to handle these, what happens if scale_factor is changed of the window?